src/app/components/post/view/view.ts

```diff
@@ -81,4 +96,68 @@ export default class AppPostView extends Vue {
 			message: this.$gettext('Your post has been published.'),
 		});
 	}
+
+	mounted() {
+		this.listenGrid();
+	}
+
+	private listenGrid() {
+		// Check every 2 seconds if we are connected to grid.
+		// Grid bootstraps async to the site load, so this page might be mounted before grid finished loading.
+		// Once connected, start listening to the post's channel.
+		this.startListenInterval = setInterval(() => {
+			if (this.startedListeningGrid) {
+				if (this.startListenInterval) {
+					clearInterval(this.startListenInterval);
+					this.startListenInterval = null;
+				}
+				return;
+			}
+
+			if (this.grid && this.grid.connected) {
+				this.startedListeningGrid = true;
+				this.grid.startListenPost(this.post, this.handleGridEvent);
+			}
+		}, 2000);
```

Comment:
This is ridiculous, i think we need to be able to just queue and abort grid operations through the grid service instead of forcing anything that requires a grid subscription to set this death contraption up.